plotsearch/serializers/plot_search.py

```diff
@@ -149,30 +150,162 @@ class Meta:
             "custom_detailed_plan",
             "custom_detailed_plan_id",
             "reservation_recipients",
+            "reservation_identifier",
+            "reservation_readable_identifier",
         )
 
+    @staticmethod
+    def get_lease_id(obj):
+        if obj.plan_unit is not None:
+            identifier = obj.plan_unit.lease_area.lease.id
+        elif obj.custom_detailed_plan is not None:
+            identifier = obj.custom_detailed_plan.lease_area.lease.id
+        else:
+            identifier = None
+        return identifier
+
+    @staticmethod
+    def get_lease_identifier(obj):
+        if obj.plan_unit is not None:
+            identifier = obj.plan_unit.lease_area.lease.identifier.identifier
+        elif obj.custom_detailed_plan is not None:
+            identifier = obj.custom_detailed_plan.lease_area.lease.identifier.identifier
+        else:
+            identifier = None
+        return identifier
+
+    @staticmethod
+    def get_lease_hitas(obj):
+        if (
+            obj.plan_unit is not None
+            and obj.plan_unit.lease_area.lease.hitas is not None
+        ):
+            hitas = obj.plan_unit.lease_area.lease.hitas.name
+        elif (
+            obj.custom_detailed_plan is not None
+            and obj.custom_detailed_plan.lease_area.lease.hitas is not None
+        ):
+            hitas = obj.custom_detailed_plan.lease_area.lease.hitas.name
+        else:
+            hitas = None
+        return hitas
+
     def get_lease_address(self, obj):
-        if obj.plan_unit is None:
-            return None
-        lease_address = (
-            obj.plan_unit.lease_area.addresses.all()
-            .order_by("-is_primary")
-            .values("address")
-            .first()
-        )
+        if obj.plan_unit is not None:
+            lease_address = (
+                obj.plan_unit.lease_area.addresses.all()
+                .order_by("-is_primary")
+                .values("address")
+                .first()
+            )
+        elif obj.custom_detailed_plan is not None:
+            lease_address = (
+                obj.custom_detailed_plan.lease_area.addresses.all()
+                .order_by("-is_primary")
+                .values("address")
+                .first()
+            )
+        else:
+            lease_address = None
         return lease_address
 
+    @staticmethod
+    def get_lease_financing(obj):
+        if (
+            obj.plan_unit is not None
+            and obj.plan_unit.lease_area.lease.financing is not None
+        ):
+            financing = obj.plan_unit.lease_area.lease.financing.name
+        elif (
+            obj.custom_detailed_plan is not None
+            and obj.custom_detailed_plan.lease_area.lease.financing is not None
+        ):
+            financing = obj.custom_detailed_plan.lease_area.lease.financing.name
+        else:
+            financing = None
+        return financing
+
+    @staticmethod
+    def get_lease_management(obj):
```

Comment:
Would it be wiser to have a function for the custom_detailed_plan vs plan unit check and use that for these serializer methodfields rather than copy paste the logic for every method?

Reply:
Well maybe there could be some idea. Nevertheless there should not be those SerializerMethodsFields almost at all. I think there should be huge reformation of accessing those fields through CustomDetailedPlanSerializer and PlanUnitSerializer